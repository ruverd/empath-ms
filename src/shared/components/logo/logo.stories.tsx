import type { Meta, StoryObj } from "@storybook/react";
import { Logo } from "./logo";

const meta = {
  title: "Components/Logo",
  component: Logo,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof Logo>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
