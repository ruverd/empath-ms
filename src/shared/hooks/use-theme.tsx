import { useContext } from "react";

import { ThemeProviderContext } from "@/shared/contexts/theme.context";

export const useTheme = () => {
  const context = useContext(ThemeProviderContext);

  if (context === undefined)
    throw new Error("useTheme must be used within a ThemeProvider");

  return context;
};
