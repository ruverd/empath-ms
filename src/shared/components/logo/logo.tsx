import brandLogoDark from "@/assets/brand/logo-dark.svg";
import brandLogoLight from "@/assets/brand/logo-light.svg";
import { useTheme } from "@/shared/hooks/use-theme";

export const Logo = () => {
  const { theme } = useTheme();

  const logoUrl = theme === "dark" ? brandLogoDark : brandLogoLight;

  return <img src={logoUrl} alt="Brand Logo" />;
};
