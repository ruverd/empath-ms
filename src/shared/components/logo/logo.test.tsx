import * as useThemeHook from "@/shared/hooks/use-theme";
import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import { Logo } from "./logo";

vi.mock("@/assets/brand/logo-dark.svg", () => ({ default: "dark-logo.svg" }));
vi.mock("@/assets/brand/logo-light.svg", () => ({ default: "light-logo.svg" }));

describe("Logo", () => {
  it("should renders light theme logo by default", () => {
    vi.spyOn(useThemeHook, "useTheme").mockReturnValue({
      theme: "light",
      setTheme: vi.fn(),
    });

    render(<Logo />);

    const logo = screen.getByAltText("Brand Logo");
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute("src", "light-logo.svg");
  });

  it("should renders dark theme logo when theme is dark", () => {
    vi.spyOn(useThemeHook, "useTheme").mockReturnValue({
      theme: "dark",
      setTheme: vi.fn(),
    });

    render(<Logo />);

    const logo = screen.getByAltText("Brand Logo");
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute("src", "dark-logo.svg");
  });

  it("has correct alt text", () => {
    vi.spyOn(useThemeHook, "useTheme").mockReturnValue({
      theme: "light",
      setTheme: vi.fn(),
    });

    render(<Logo />);

    expect(screen.getByAltText("Brand Logo")).toBeInTheDocument();
  });

  it("matches light theme snapshot", () => {
    vi.spyOn(useThemeHook, "useTheme").mockReturnValue({
      theme: "light",
      setTheme: vi.fn(),
    });

    const { container } = render(<Logo />);
    expect(container).toMatchSnapshot();
  });

  it("matches dark theme snapshot", () => {
    vi.spyOn(useThemeHook, "useTheme").mockReturnValue({
      theme: "dark",
      setTheme: vi.fn(),
    });

    const { container } = render(<Logo />);
    expect(container).toMatchSnapshot();
  });
});
