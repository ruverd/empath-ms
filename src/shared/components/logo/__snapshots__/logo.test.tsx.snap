// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo > matches dark theme snapshot 1`] = `
<div>
  <img
    alt="Brand Logo"
    src="dark-logo.svg"
  />
</div>
`;

exports[`Logo > matches light theme snapshot 1`] = `
<div>
  <img
    alt="Brand Logo"
    src="light-logo.svg"
  />
</div>
`;
